app/src/main/java/ai/elimu/vitabu/ui/StoryBooksActivity.java

```diff
@@ -78,14 +83,9 @@ public void run() {
                     runOnUiThread(new Runnable() {
                         @Override
                         public void run() {
-                            File imageFile = new File(Environment.getExternalStorageDirectory() +
-                                    "/Android/data/" +
-                                    BuildConfig.CONTENT_PROVIDER_APPLICATION_ID +
-                                    "/files/" + Environment.DIRECTORY_PICTURES + "/" +
-                                    coverImage.getId() + "_r" + coverImage.getRevisionNumber() + "." + coverImage.getImageFormat().toString().toLowerCase());
-                            Uri imageFileUri = Uri.fromFile(imageFile);
-                            Log.i(getClass().getName(), "imageFileUri: " + imageFileUri);
-                            coverImageView.setImageURI(imageFileUri);
+                            String imageFilePath = PICTURES_PATH + coverImage.getId() + "_r" + coverImage.getRevisionNumber() + "." + coverImage.getImageFormat().toString().toLowerCase();
+                            Log.i(getClass().getName(), "imageFile: " + imageFilePath);
+                            coverImageView.setImageBitmap(decodeSampledBitmap(imageFilePath, 100, 100));
```

Comment:
This really cool to know that we can load a scaled-down version of the image into memory. Can we use this even to load images associated with each chapter once the storybook is opened? Just wondering if we can test 100 * 100 px would be good enough for any device using the mobile app version (say tablet) or we can go even lower?

Reply:
@Keerthi4308 Actually, when images are extracted from a storybook ePUB (or uploaded manually), they are scaled down to 640px before being stored in the webapp's database: https://github.com/elimu-ai/webapp/blob/master/src/main/java/ai/elimu/util/ImageHelper.java

Once the storybook is opened, these original (not compressed) 640px images are displayed to the child in the storybook (Vitabu) app and in other Android apps.

![device-2020-11-29-204051](https://user-images.githubusercontent.com/15718174/100551777-949b9800-328b-11eb-8798-2496bc6962c0.png)

Our goal is to come up with a solution that works well on all display sizes ranging from 6" (smartphones) to 10" (tablets). Perhaps storing multiple variants (resolutions) of the same image in the webapp's database would be useful?